kernel-specification.yml

```diff
@@ -35,19 +35,19 @@ kernels:
               for i in range(bitmasklength):
                   byte = frombitmask[i]
                   tobytemask[(i * 8) + 0] = ((byte & uint8(128)) != 0) != validwhen
-                  byte <<= 1
+                  byte = (byte << 1) & 0xFF
```

Comment:
That makes sense: a hardware `<<` instruction would clamp to the integer size automatically (because there's nowhere else for the bits to go), but Python's software `<<` instructions are implemented in arbitrary-precision arithmetic, so the "byte" can hold a value bigger than 255.

I'm sure that this was intended as a convenience, but it means we need to be extra careful when applying any bitwise operation (`<<`, `>>`, `&`, `|`, `~`) to a Python integer. Results of bitwise operations can depend on bit width (consider `~`, especially) but the notion of "bit width" can be nebulous if Python's arbitrarily increasing it to fit the integer value.

In normal Python code, I try to avoid this by casting integers as NumPy integers with a specific width, such as `np.uint8` or `np.int64`. In the kernel specification code, however, we should use only basic Python so what you have here is ideal.